safe-discriminant-derive/src/lib.rs

```diff
@@ -80,13 +80,30 @@ fn validate_all_variants(variants: impl Iterator<Item = Variant>) -> Result<()>
         .unwrap_or(Ok(()))
 }
 
+/// Returns true if there is any #[x] where x is not `repr`
+/// returns false otherwise.
+fn contains_attribute_macros(attrs: &[Attribute]) -> bool {
+    attrs
+        .iter()
+        .filter(|attr| !attr.path().is_ident("repr"))
+        .count()
```

Comment:
Nit: Probably `any()` is a better use case for this: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any

```rs
attrs.iter().any(|attr| !attr.path().is_ident("repr"))
```

Side-Note: I actually didn't know that the derive macro can't see the derive attributes. TIL.

Reply:
> Side-Note: I actually didn't know that the derive macro can't see the derive attributes. TIL.

Apparently i is a bit tricky, they cannot see derive attribute, and (I think) they cannot see attributed defined before them if they did not persist after the attribute is being called.